dmscripts/notify_suppliers_of_brief_withdrawal.py

```diff
@@ -0,0 +1,45 @@
+from dmutils.email.helpers import hash_string
+
+from dmscripts.helpers import env_helpers
+
+
+def get_brief_response_emails(data_api_client, brief_id):
+    responses = data_api_client.find_brief_responses(brief_id=brief_id, status="submitted").get("briefResponses")
+    return [response["respondToEmailAddress"] for response in responses]
+
+
+def create_context_for_brief(stage, brief):
+    return {
+        'brief_title': brief['title'],
+        'brief_link': '{0}/{1}/opportunities/{2}'.format(
+            env_helpers.get_web_url_from_stage(stage),
+            brief['frameworkFramework'],
+            brief['id']
+        )
+    }
+
+
+def main(data_api_client, mail_client, template_id, stage, logger, withdrawn_date=None, brief_id=None, dry_run=False):
```

Comment:
Should `withdrawn_date` be a kwarg like this? If for some reason it's not given a value, you'll end up with ALL briefs, not just withdrawn.

The other part of the script that does the parsing of arguments etc sets `withdrawn_at` to yesterday if it's not already set, so this could be a positional argument so that it's harder to pass in None by accident.

I might be being too paranoid though....

Reply:
Great shout (I'll probably have nightmares about that scenario tonight 😂 )

Reply:
You could add a guard clause in at the top of the script instead of changing it to a positional arg. Something like?
```
if not withdrawn_on:
    Abort
```

Reply:
Yeah I'll add it to the request to the api

Reply:
In the api client?